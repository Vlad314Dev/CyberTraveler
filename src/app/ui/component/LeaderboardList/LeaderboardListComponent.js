import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import './LeaderboardListStyle.scss';

class LeaderboardListComponent extends PureComponent
{
    static propTypes = {
        loading: PropTypes.bool.isRequired,
        data: PropTypes.object.isRequired,
        error: PropTypes.string
    };

    render()
    {
        const { 
            loading, error, data 
        } = this.props;
        
        if (loading) {
            return (
                <div block="LeaderboardList">Loading...</div>
            );
        }

        if (error) {
            return (
                <div block="LeaderboardList">Error: { error }</div>
            )
        }

        if (data) {
            return (
                <div block="LeaderboardList">
                    Leaderboard
                    {
                        data.getUsers.map(( user, idx ) => (
                            <p key={ idx }>{ user.nickname }</p>
                        ))
                    }
                </div>
            )
        }
    }
}

export default LeaderboardListComponent;
