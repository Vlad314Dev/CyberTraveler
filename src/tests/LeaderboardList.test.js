/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import LeaderboardListComponent from '../app/ui/component/LeaderboardList/LeaderboardListComponent';

describe('LeaderboardList', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <LeaderboardListComponent 
                loading={ false }
                data={ {
                    getLeaderboard: []
                } }
                error={ '' }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <LeaderboardListComponent 
                loading={ false }
                data={ {
                    getLeaderboard: []
                } }
                error={ '' }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
