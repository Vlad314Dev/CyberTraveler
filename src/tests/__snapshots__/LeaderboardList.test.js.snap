// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardList Should render layout 1`] = `
Array [
  <div
    className="LeaderboardList"
  >
    <div
      className="LeaderboardList__Content"
    >
      <table>
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Player
            </th>
            <th>
              Score
            </th>
            <th>
              Date
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
]
`;
